File: app/src/main/java/com/projetdev/malo/musichall/models/Album.java
package com.projetdev.malo.musichall.models;

import java.util.ArrayList;
import java.util.Map;

public class Album extends Item {

    private ArrayList<Track> tracks;
    private ArrayList<String> tags;
    private String summup;
    private String content;

    public Album(String id, String name, String url, Map<String, String> images) {
        super(id, name, url, images);
    }

    public Album(String id, String name, String url, Integer year, Artist artist, Map<String, String> images) {
        super(id, name, url, images, year, artist);
    }

    public Album(String id, String name, String url, Integer year, Artist artist, Map<String, String> images, ArrayList<Track> tracks, ArrayList<String> tags, String summup, String content) {
        super(id, name, url, images, year, artist);
        this.tracks = tracks;
        this.tags = tags;
        this.summup = summup;
        this.content = content;
    }


    public ArrayList<Track> getTracks() {
        return tracks;
    }

    public void setTracks(ArrayList<Track> tracks) {
        this.tracks = tracks;
    }

    public ArrayList<String> getTags() {
        return tags;
    }

    public void setTags(ArrayList<String> tags) {
        this.tags = tags;
    }

    public String getSummup() {
        return summup;
    }

    public void setSummup(String summup) {
        this.summup = summup;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: app/src/main/java/com/projetdev/malo/musichall/models/Album.kt
package com.projetdev.malo.musichall.models

class Album(
    override val id:        String,
    override val name:      String,
    override val url:       String,
    val artist:             Artist,
    override val images:    Map<String,String>
) : Item(id, name, url, images) {
    constructor(
        id:         String,
        name:      String,
        url:        String,
        artist:     Artist,
        images:     Map<String,String>,
        tracks:     ArrayList<Track>?,
        tags:       ArrayList<String>?,
        summup:     String?,
        content:    String?
    ) : this(id, name, url, artist, images)
}